test/fixtures/programmatic/plugin/extend-config-plugin/index.js

```diff
@@ -0,0 +1,35 @@
+'use strict';
+
+module.exports = class TestPlugin {
+  constructor(serverless, options, utils) {
+    this.serverless = serverless;
+    this.options = options;
+    this.utils = utils;
+
+    this.target = this.serverless.configurationInput.custom.extendConfig.target;
+    this.value = this.serverless.configurationInput.custom.extendConfig.value;
+    if (typeof this.value === 'string') {
+      this.value = this.value.replace(/%/g, '$');
+    }
+
+    this.hook = this.serverless.configurationInput.custom.extendConfig.hook;
+    if (this.hook !== undefined) {
+      this.hooks = {
+        [this.hook]: () => this.extend(),
+      };
+    } else {
+      this.hook = 'async init';
+    }
+  }
+
+  async asyncInit() {
+    if (this.hooks === undefined) {
+      this.extend();
+    }
+  }
```

Comment:
It'll be nicer if we simplify that logic. So plugin just added some variables to config in `asyncInit`, in most straightforward way

I understand that what's proposed tries to recreate some plugin idea, but this makes this test more complex and more difficult to comprehend than it should be.

Ideally, if any developer just by looking at it in the first seconds, has no doubt what the action made by this plugin.

And we're purely testing variable extensions, nothing else

Reply:
Done.